### ISSUE TITLE
Lock down S3 buckets

### ISSUE THREAD

**Issue by kurtwheeler** [AlexsLemonade/refinebio#731]
### Context

It's possible to download objects from our S3 buckets if you the URL to the object, despite the permissions to list the contents of S3 buckets being locked down.

### Problem or idea

We should make it so that you need to have permissions to the data to download it, for legal reasons.

### Solution or next step

However will want people to be able to pull data from those buckets eventually, a prime example of this is transcriptome indices. I think this is the strategy @Miserlou mentioned for enabling that: https://docs.aws.amazon.com/STS/latest/APIReference/Welcome.html

### New Issue Checklist

- [x] The title is short and descriptive.
- [x] You have explained the context that led you to write this issue.
- [x] You have reported a problem or idea.
- [x] You have proposed a solution or next step.

**Comment by kurtwheeler** [comment]
@cgreene I've been digging into this one a bit, what I'm thinking is setting all our buckets to private and then using https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3.html#generating-presigned-urls to generate a URL that will be valid for only a few hours to allow users to download data. This way they can get whatever we want to let them at, but only after they've agreed to our T&C. Will this satisfy the lawyers?

However, there's a question about what data needs to be locked down in this manner:
```
kurt-al [4:24 PM]
OK got it. Unsmashed processed data, downloadable datasets, and compendia should be private, while QN targets and tx-indices are public but only discoverable via API

jackie-al [4:25 PM]
yea I suspect those that would want those things are sufficiently computationally savvy
we may want a page that's like "here's some other stuff to make things work" eventually

kurt-al [4:26 PM]
there won't be legal concerns around distributing qn targets/tx-indices?

jackie-al [4:26 PM]
shouldn't be
I don't think one could figure out the original values of the samples used to derive the QN targets
and the transcriptome index doesn't have anything to do with individuals

kurt-al [4:27 PM]
What about in the "we used your data to make a decision and it was wrong" category?

jackie-al [4:28 PM]
oh hm
```

I can't remember if those are actual concerns or if I just overheard them as something that might be an issue. Which types of S3 objects do we need to make private and/or are there any that can be public?

This change will mean anything we put in a private bucket won't be accessible unless there's a webpage for it which allows you to agree to T&C and get a URL. At least not until we implement a way to agree via API.

**Comment by cgreene** [comment]
The risk for QN targets / tx-indices would seem to be very small. I think the main concern is data for specific samples. At this point, I'm comfortable using the restricted buckets for the dataset downloads but not other items.

**Comment by arielsvn** [comment]
With https://github.com/AlexsLemonade/refinebio/issues/2236 we discovered that the computed files are currently publicly available. I'm going to lock that bucket as well as add additional documentation to the API on how to download them sending a valid Token.
